#! /bin/sh /usr/share/dpatch/dpatch-run
## 06keygen.dpatch by  <bernat@neo.luffy.cx>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Complete de and fr layout

@DPATCH@

diff -b lang.c.old lang.c
--- xrdp-0.4.0.orig/xrdp/lang.c	2008-02-02 00:44:45.000000000 +0100
+++ xrdp-0.4.0/xrdp/lang.c	2008-02-02 00:48:14.000000000 +0100
@@ -293,7 +293,7 @@
 char de_noshift[] =
 {
   0,    0,   '1',  '2',  '3',  '4',  '5',  '6',
- '7',  '8',  '9',  '0',  0xdf, 0xb4,  0,    0,
+ '7',  '8',  '9',  '0',  0xdf,  0,    0,    0,  
  'q',  'w',  'e',  'r',  't',  'z',  'u',  'i',
  'o',  'p',  0xfc, '+',   0,    0,   'a',  's',
  'd',  'f',  'g',  'h',  'j',  'k',  'l',  0xf6,
@@ -302,7 +302,7 @@
   0,   ' ',   0,    0,    0,    0,    0,    0,
   0,    0,    0,    0,    0,    0,    0,   '7',
  '8',  '9',  '-',  '4',  '5',  '6',  '+',  '1',
- '2',  '3',  '0',  '.',   0,    0,    0,    0,
+ '2',  '3',  '0',  '.',   0,    0,   '<',   0,  
   0,    0,    0,    0,    0,    0,    0,    0,
   0,    0,    0,    0,    0,    0,    0,    0,
   0,    0,    0,    0,    0,    0,    0,    0,
@@ -314,7 +314,7 @@
 char de_shift[] =
 {
   0,    0,   '!',  '"',  0xa7, '$',  '%',  '&',
- '/',  '(',  ')',  '=',  '?',  '`',   0,    0,
+ '/',  '(',  ')',  '=',  '?',   0,    0,    0,  
  'Q',  'W',  'E',  'R',  'T',  'Z',  'U',  'I',
  'O',  'P',  0xdc, '*',   0,    0,   'A',  'S',
  'D',  'F',  'G',  'H',  'J',  'K',  'L',  0xd6,
@@ -323,7 +323,7 @@
   0,   ' ',   0,    0,    0,    0,    0,    0,
   0,    0,    0,    0,    0,    0,    0,   '7',
  '8',  '9',  '-',  '4',  '5',  '6',  '+',  '1',
- '2',  '3',  '0',  '.',   0,    0,    0,    0,
+ '2',  '3',  '0',  '.',   0,    0,   '>',   0,  
   0,    0,    0,    0,    0,    0,    0,    0,
   0,    0,    0,    0,    0,    0,    0,    0,
   0,    0,    0,    0,    0,    0,    0,    0,
@@ -334,17 +334,17 @@
 /* right alt chars */
 char de_altgr[] =
 {
-  0,    0,    0,   0xb2, 0xb3,  0,    0,    0,
+  0,    0,   0xb9, 0xb2, 0xb3, 0xbc, 0xbd, 0xbe,
  '{',  '[',  ']',  '}',  '\\',  0,    0,    0,
- '@',   0,   0x80,  0,    0,    0,    0,    0,
-  0,    0,    0,   '~',   0,    0,    0,    0,
-  0,    0,    0,    0,    0,    0,    0,    0,
-  0,    0,    0,    0,    0,    0,    0,    0,
-  0,    0,   0xb5,  0,    0,    0,    0,    0,
-  0,    0,    0,    0,    0,    0,    0,    0,
+ '@',   0,    0,   0xb6,  0,    0,    0,    0,  
+ 0xf8, 0xfe,  0,    0,    0,    0,   0xe6,  0,  
+ 0xf0,  0,    0,    0,    0,    0,    0,    0,  
+  0,   0xac,  0,    0,   0xab, 0xbb, 0xa2,  0,  
+  0,    0,   0xb5,  0,   0xb7,  0,    0,    0,  
   0,    0,    0,    0,    0,    0,    0,    0,
   0,    0,    0,    0,    0,    0,    0,    0,
   0,    0,    0,    0,    0,    0,    0,    0,
+  0,    0,    0,    0,    0,    0,   '|',   0,  
   0,    0,    0,    0,    0,    0,    0,    0,
   0,    0,    0,    0,    0,    0,    0,    0,
   0,    0,    0,    0,    0,    0,    0,    0,
@@ -368,12 +368,12 @@
   0,   ' ',   0,    0,    0,    0,    0,    0,
   0,    0,    0,    0,    0,    0,    0,   '7',
  '8',  '9',  '-',  '4',  '5',  '6',  '+',  '1',
- '2',  '3',  '0',  '.',   0,    0,    0,    0,
+ '2',  '3',  '0',  '.',   0,    0,   '<',   0,  
+  0,    0,    0,    0,    0,    0,    0,    0,  
   0,    0,    0,    0,    0,    0,    0,    0,
   0,    0,    0,    0,    0,    0,    0,    0,
   0,    0,    0,    0,    0,    0,    0,    0,
   0,    0,    0,    0,    0,    0,    0,    0,
-  0,    0,    0,    0,    0,    0,    0,    0
 };
 
 /* shift chars */
@@ -382,40 +382,40 @@
   0,    0,   '1',  '2',  '3',  '4',  '5',  '6',
  '7',  '8',  '9',  '0',  0xb0, '+',   0,    0,
  'A',  'Z',  'E',  'R',  'T',  'Y',  'U',  'I',
- 'O',  'P',  0xa8, 0xa3,  0,    0,   'Q',  'S',
+ 'O',  'P',  '"',  0xa3,  0,    0,   'Q',  'S', 
  'D',  'F',  'G',  'H',  'J',  'K',  'L',  'M',
  '%',  '~',   0,   0xb5, 'W',  'X',  'C',  'V',
  'B',  'N',  '?',  '.',  '/',  0xa7,  0,   '*',
   0,   ' ',   0,    0,    0,    0,    0,    0,
   0,    0,    0,    0,    0,    0,    0,   '7',
  '8',  '9',  '-',  '4',  '5',  '6',  '+',  '1',
- '2',  '3',  '0',  '.',   0,    0,    0,    0,
+ '2',  '3',  '0',  '.',   0,    0,   '>',   0,  
+  0,    0,    0,    0,    0,    0,    0,    0,  
   0,    0,    0,    0,    0,    0,    0,    0,
   0,    0,    0,    0,    0,    0,    0,    0,
   0,    0,    0,    0,    0,    0,    0,    0,
   0,    0,    0,    0,    0,    0,    0,    0,
-  0,    0,    0,    0,    0,    0,    0,    0
 };
 
 /* right alt chars */
 char fr_altgr[] =
 {
-  0,    0,    0,   '~',  '#',  '{',  '[',  '|',
+  0,    0,   0xb9, '~',  '#',  '{',  '[',  '|', 
  '`',  '\\', '^',  '@',  ']',  '}',   0,    0,
-  0,    0,   0x80,  0,    0,    0,    0,    0,
-  0,    0,    0,   0xa4,  0,    0,    0,    0,
-  0,    0,    0,    0,    0,    0,    0,    0,
-  0,    0,    0,    0,    0,    0,    0,    0,
+ 0xe6, 0xab,  0,   0xb6,  0,    0,    0,    0,  
+ 0xf8, 0xfe,  0,   0xa4,  0,    0,    0,   0xdf,
+ 0xf0,  0,    0,    0,    0,    0,    0,    0,  
+  0,   0xac,  0,    0,    0,   0xbb, 0xa2,  0,  
+  0,    0,    0,    0,   0xb7,  0,    0,    0,  
   0,    0,    0,    0,    0,    0,    0,    0,
   0,    0,    0,    0,    0,    0,    0,    0,
   0,    0,    0,    0,    0,    0,    0,    0,
+  0,    0,    0,    0,    0,    0,   '|',   0,  
   0,    0,    0,    0,    0,    0,    0,    0,
   0,    0,    0,    0,    0,    0,    0,    0,
   0,    0,    0,    0,    0,    0,    0,    0,
   0,    0,    0,    0,    0,    0,    0,    0,
   0,    0,    0,    0,    0,    0,    0,    0,
-  0,    0,    0,    0,    0,    0,    0,    0,
-  0,    0,    0,    0,    0,    0,    0,    0
 };
 
 /*****************************************************************************/
